### ISSUE TITLE
Proposal to change end of maintenance timetable for JupyterLab

### ISSUE THREAD

**Issue by JasonWeill** [jupyterlab/team-compass#231]
### Problem

Per @jasongrout 's [comment](https://github.com/jupyterlab/jupyterlab/issues/12397#issuecomment-1414393697) in #12397, we will support a major version of JupyterLab for 1 year after its successor's first release. In the case of JupyterLab 3, JupyterLab 4.0.0 was released on May 15, 2023, so JupyterLab 3 will no longer be maintained after May 15, 2024.

Should we adopt this schedule? If so, we will need to both update our documentation and use all our social channels, including the blog and forum, to publicize this change.

### Proposed Solution

https://github.com/jupyterlab/jupyterlab/pull/15626 updates the documentation to reflect the new end of maintenance schedule.

### Additional context

https://github.com/jupyterlab/jupyterlab/pull/12829, fixing https://github.com/jupyterlab/jupyterlab/issues/12828, documents our "current and previous major version" support lifecycle. Our current lifecycle suggests that JupyterLab 3 will be maintained until JupyterLab 5.0.0 is released.

**Comment by jtpio** [comment]
cc @jupyterlab/jupyterlab-council for awareness

**Comment by JasonWeill** [comment]
We discussed the proposed end of maintenance timetable for JupyterLab 3. If we announced this very soon, that would be roughly 3.5 months' advance notice that Lab 3 would be ending maintenance effective May 15. Summarized notes from the weekly JupyterLab call (thanks @jtpio and @krassowski for contributing):

* Suggestion from @krassowski: communicate the May 15 date externally as a question/suggestion rather than a "this is what we're doing"
* Allow possibility for extending the deadline by 3 or 6 months
* [Download numbers v3 vs v4](https://www.pepy.tech/projects/jupyterlab?versions=4.*&versions=3.*) — note that Lab 4 has consistently 5 to 10 times as many downloads as Lab 3 does, and that we still have tens of thousands of Lab 3 downloads on a typical weekday. This also suggests that many users are still actively using Lab 3, since users don't need to download the software every time they use it.
  * download numbers for 3.6.x should be as representative as they get since we just had a patch release and users should have been notified
